### PR TITLE
port to dune and port opam metadata to 2.0

### PR DESCRIPTION
**Pull request by avsm**
* Port library to Dune from jbuilder and the builtin
  `dune.configurator` to reduce the build dependency cone
* Remove unused variable warnings (@avsm).
* Update opam package metadata to 2.0 format (@avsm).

### CODE REVIEW

**Comment by avsm** [comment]
This should be good to merge if someone can review it first.

File: unix/mclock.ml [hunk 1]
```diff
@@ -15,7 +15,7 @@
  *)
 type t = unit
 type 'a io = 'a Lwt.t
-type error = unit
+type _error = unit
```
Comment (dinosaure) [inline unix/mclock.ml]: I prefer to put `[@@@warning "-34"]` or it's possible to just delete it?
NOTE: `type[@warning "-34"] error` does not work ... I will make an issue about that in OCaml.

File: config/dune [hunk 2]
```diff
@@ -0,0 +1,8 @@
+(executables
+ (names discover)
+ (libraries dune.configurator))
+
+(rule
+ (targets cclib.sexp)
+ (deps    discover.exe)
+ (action  (run ./discover.exe)))
```
Comment (dinosaure) [inline config/dune:8]: I prefer to have something like:

```
 (deps (:exe discover.exe))
 (action (run %{exe}))
```

But it's ok.

Comment (avsm) [reply config/dune:8]: I wasnt quite sure how that would interact with having multiple `discover.exe` in the same duniverse repo..

File: unix/mclock.ml (hunk 1 above)
Comment (avsm) [reply unix/mclock.ml]: removed it in bbaaa918277463cfe366d4b811f73456a7a853e4 indeed :)

**Comment by dinosaure** [comment]
Ok to merge for me.

**Comment by avsm** [comment]
Thanks for the review!

**Pull request merged**